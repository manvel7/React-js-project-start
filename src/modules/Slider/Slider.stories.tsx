import React from "react"
import {Meta, Story} from "@storybook/react/types-6-0"
import Slider from "./Slider"

const meta: Meta = {
  title: "Modules/Slider",
  component: Slider
}

const SliderStory: Story = (args) => (
  <Slider {...args} slides={[<div key="1">1</div>, <div key="2">2</div>]}>
    slider
  </Slider>
)

const Default = SliderStory.bind({})

export {Default}

export default meta
