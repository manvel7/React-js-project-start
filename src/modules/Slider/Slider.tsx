import React, {
  FunctionComponent,
  HTMLAttributes,
  ReactElement,
  useState
} from "react"
import {Pagination} from "../../index"
import GhostWrapper from "./GhostWrapper"
import SliderItem from "./Item"
import PaginationWrapper from "./PaginationWrapper"
import SliderWrapper from "./Wrapper"

interface IProps extends HTMLAttributes<HTMLDivElement> {
  slides: ReactElement[]
}

const Slider: FunctionComponent<IProps> = (props) => {
  const {slides} = props

  const [currentSlide, setCurrentSlide] = useState<number>(1)

  const computedSlides = React.Children.map(slides, (slide, index) => {
    return (
      <SliderItem isActive={index + 1 === currentSlide}>{slide}</SliderItem>
    )
  })

  return (
    <div>
      <SliderWrapper>
        {computedSlides}
        <GhostWrapper>{slides}</GhostWrapper>
      </SliderWrapper>

      <PaginationWrapper>
        <Pagination
          variant="circle"
          hidePageNumber={true}
          currentPage={currentSlide}
          pageCount={slides.length}
          onPageChange={setCurrentSlide}
        />
      </PaginationWrapper>
    </div>
  )
}

export default Slider
