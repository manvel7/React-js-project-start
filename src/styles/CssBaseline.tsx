import {
  createGlobalStyle,
  css,
  DefaultTheme,
  GlobalStyleComponent,
  StyledProps
} from "styled-components"
import {mediaBreakpointDown} from "../utils/breakpoint"
import {getValue, toPercent} from "../utils/unit"
import {
  bodyMixin,
  smallMixin,
  h1Mixin,
  h2Mixin,
  h3Mixin,
  h4Mixin,
  h5Mixin,
  h6Mixin
} from "./typography"

interface IProps {}

const genericResetMixin = css<StyledProps<IProps>>`
  html,
  body,
  div,
  span,
  applet,
  object,
  iframe,
  h1,
  h2,
  h3,
  h4,
  h5,
  h6,
  p,
  blockquote,
  pre,
  a,
  abbr,
  acronym,
  address,
  big,
  cite,
  code,
  del,
  dfn,
  em,
  img,
  ins,
  kbd,
  q,
  s,
  samp,
  small,
  strike,
  strong,
  sub,
  sup,
  tt,
  var,
  b,
  u,
  i,
  center,
  dl,
  dt,
  dd,
  ol,
  ul,
  li,
  fieldset,
  form,
  label,
  legend,
  table,
  caption,
  tbody,
  tfoot,
  thead,
  tr,
  th,
  td,
  article,
  aside,
  canvas,
  details,
  embed,
  figure,
  figcaption,
  footer,
  header,
  hgroup,
  menu,
  nav,
  output,
  ruby,
  section,
  summary,
  time,
  mark,
  audio,
  video {
    margin: 0;
    padding: 0;
    border: 0;
    vertical-align: baseline;
  }

  article,
  aside,
  details,
  figcaption,
  figure,
  footer,
  header,
  hgroup,
  menu,
  nav,
  section {
    display: block;
  }

  ol,
  ul {
    list-style: none;
  }

  blockquote,
  q {
    quotes: none;
  }

  blockquote:before,
  blockquote:after,
  q:before,
  q:after {
    content: "";
    content: none;
  }

  table {
    border-collapse: collapse;
    border-spacing: 0;
  }
`

const appResetMixin = css<StyledProps<IProps>>`
  html {
    /* stylelint-disable property-no-vendor-prefix */
    -ms-overflow-style: scrollbar; /* Set box-sizing so that the width is not affected by padding or border */
    -ms-text-size-adjust: 100%; /* Set default font family in all browsers. */
    -webkit-tap-highlight-color: rgba(0, 0, 0, 0); /* Makes 1rem = 10px */
    -webkit-text-size-adjust: 100%; /* Prevent adjustments of font size after orientation changes in IE on Windows Phone and in iOS. */
    /* stylelint-enable property-no-vendor-prefix */
    box-sizing: border-box;
  }

  *,
  *::before,
  *::after {
    box-sizing: inherit;
  }

  body {
    background-color: ${({theme}) => theme.palette.background.white};

    /* Hide overflow content */
    &.overflow {
      overflow: hidden;
    }

    /* Prevent focus outline on elements that can't be accessed via keyboard. */
    [tabindex="-1"]:focus {
      outline: none !important;
    }

    /* Hide overflow in IE */
    svg:not(:root) {
      overflow: hidden;
    }

    /* Hide elements has 'hidden' attribute in IE 10- */
    [hidden] {
      display: none !important;
    }

    ::-webkit-file-upload-button {
      /* stylelint-disable-next-line property-no-vendor-prefix */
      -webkit-appearance: button;
      font: inherit;
    }
  }
`

const typographyMixin = css<StyledProps<IProps>>`
  html {
    font-family: ${({theme}) =>
      theme.typography
        .fontFamily}; /* Prevent scrollbars to overlap content in IE11 and Edge */
    font-size: ${({theme}) =>
      toPercent(
        getValue(theme.typography.htmlFontSize),
        16
      )}; /* Set default tap highlight to be transparent in iOS. */
    font-style: normal;
    color: ${({theme}) => theme.palette.text.dark};
  }

  body {
    ${bodyMixin<IProps>()}
  }

  h1 {
    ${h1Mixin<IProps>()}
  }

  h2 {
    ${h2Mixin<IProps>()}
  }

  h3 {
    ${h3Mixin<IProps>()}
  }

  h4 {
    ${h4Mixin<IProps>()}
  }

  h5 {
    ${h5Mixin<IProps>()}
  }

  h6 {
    ${h6Mixin<IProps>()}
  }

  small {
    ${smallMixin<IProps>()}
  }

  b,
  strong {
    font-weight: ${({theme}) => theme.typography.fontWeights.bold};

    h1,
    h2,
    h3,
    h4,
    h5,
    h6,
    caption {
      font-weight: ${({theme}) => theme.typography.fontWeights.boldest};
    }
  }
`

const utilsMixin = css<StyledProps<IProps>>`
  .hidden {
    visibility: hidden;
  }

  .d-none {
    display: none;
  }

  .hidden-sm-down {
    ${({theme}) =>
      mediaBreakpointDown(
        "sm",
        theme.grid.breakpoints,
        `
      display: none;
    `
      )}
  }

  .text-center {
    text-align: center;
  }
`

const CssBaseline: GlobalStyleComponent<
  IProps,
  DefaultTheme
> = createGlobalStyle`
  ${genericResetMixin}

  ${appResetMixin}

  ${typographyMixin}

  ${utilsMixin}
`

export default CssBaseline
