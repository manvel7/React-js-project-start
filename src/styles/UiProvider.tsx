import React, {FunctionComponent} from "react"
import {ThemeProvider} from "styled-components"
import theme from "./theme"
import {CssBaseline} from "./index"

interface IProps {}

const UiProvider: FunctionComponent<IProps> = (props) => {
  const {children} = props

  return (
    <ThemeProvider theme={theme}>
      <CssBaseline />
      {children}
    </ThemeProvider>
  )
}

export default UiProvider
