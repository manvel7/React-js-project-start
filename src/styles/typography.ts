import {
  css,
  DefaultTheme,
  FlattenInterpolation,
  StyledProps,
  ThemedStyledProps
} from "styled-components"
import {pxToRem} from "../utils/unit"

const h1Mixin = <P extends Record<any, any>>(): FlattenInterpolation<
  ThemedStyledProps<P, DefaultTheme>
> => css<StyledProps<P>>`
  font-size: ${({theme}) =>
    pxToRem(theme.typography.variants.h1.size, theme.typography.htmlFontSize)};
  font-weight: ${({theme}) => theme.typography.variants.h1.weight};
  letter-spacing: ${({theme}) => theme.typography.variants.h1.letterSpacing};
  line-height: ${({theme}) =>
    pxToRem(
      theme.typography.variants.h1.lineHeight,
      theme.typography.htmlFontSize
    )};

  b,
  strong {
    font-weight: ${({theme}) => theme.typography.fontWeights.boldest};
  }
`

const h2Mixin = <P extends Record<any, any>>(): FlattenInterpolation<
  ThemedStyledProps<P, DefaultTheme>
> => css<StyledProps<P>>`
  font-size: ${({theme}) =>
    pxToRem(theme.typography.variants.h2.size, theme.typography.htmlFontSize)};
  font-weight: ${({theme}) => theme.typography.variants.h2.weight};
  letter-spacing: ${({theme}) => theme.typography.variants.h2.letterSpacing};
  line-height: ${({theme}) =>
    pxToRem(
      theme.typography.variants.h2.lineHeight,
      theme.typography.htmlFontSize
    )};

  b,
  strong {
    font-weight: ${({theme}) => theme.typography.fontWeights.boldest};
  }
`

const h3Mixin = <P extends Record<any, any>>(): FlattenInterpolation<
  ThemedStyledProps<P, DefaultTheme>
> => css<StyledProps<P>>`
  font-size: ${({theme}) =>
    pxToRem(theme.typography.variants.h3.size, theme.typography.htmlFontSize)};
  font-weight: ${({theme}) => theme.typography.variants.h3.weight};
  letter-spacing: ${({theme}) => theme.typography.variants.h3.letterSpacing};
  line-height: ${({theme}) =>
    pxToRem(
      theme.typography.variants.h3.lineHeight,
      theme.typography.htmlFontSize
    )};

  b,
  strong {
    font-weight: ${({theme}) => theme.typography.fontWeights.boldest};
  }
`

const h4Mixin = <P extends Record<any, any>>(): FlattenInterpolation<
  ThemedStyledProps<P, DefaultTheme>
> => css<StyledProps<P>>`
  font-size: ${({theme}) =>
    pxToRem(theme.typography.variants.h4.size, theme.typography.htmlFontSize)};
  font-weight: ${({theme}) => theme.typography.variants.h4.weight};
  letter-spacing: ${({theme}) => theme.typography.variants.h4.letterSpacing};
  line-height: ${({theme}) =>
    pxToRem(
      theme.typography.variants.h4.lineHeight,
      theme.typography.htmlFontSize
    )};

  b,
  strong {
    font-weight: ${({theme}) => theme.typography.fontWeights.boldest};
  }
`

const h5Mixin = <P extends Record<any, any>>(): FlattenInterpolation<
  ThemedStyledProps<P, DefaultTheme>
> => css<StyledProps<P>>`
  font-size: ${({theme}) =>
    pxToRem(theme.typography.variants.h5.size, theme.typography.htmlFontSize)};
  font-weight: ${({theme}) => theme.typography.variants.h5.weight};
  letter-spacing: ${({theme}) => theme.typography.variants.h5.letterSpacing};
  line-height: ${({theme}) =>
    pxToRem(
      theme.typography.variants.h5.lineHeight,
      theme.typography.htmlFontSize
    )};

  b,
  strong {
    font-weight: ${({theme}) => theme.typography.fontWeights.boldest};
  }
`

const h6Mixin = <P extends Record<any, any>>(): FlattenInterpolation<
  ThemedStyledProps<P, DefaultTheme>
> => css<StyledProps<P>>`
  font-size: ${({theme}) =>
    pxToRem(theme.typography.variants.h6.size, theme.typography.htmlFontSize)};
  font-weight: ${({theme}) => theme.typography.variants.h6.weight};
  letter-spacing: ${({theme}) => theme.typography.variants.h6.letterSpacing};
  line-height: ${({theme}) =>
    pxToRem(
      theme.typography.variants.h6.lineHeight,
      theme.typography.htmlFontSize
    )};

  b,
  strong {
    font-weight: ${({theme}) => theme.typography.fontWeights.boldest};
  }
`

const bodyMixin = <P extends Record<any, any>>(): FlattenInterpolation<
  ThemedStyledProps<P, DefaultTheme>
> => css<StyledProps<P>>`
  font-size: ${({theme}) =>
    pxToRem(
      theme.typography.variants.body.size,
      theme.typography.htmlFontSize
    )};
  font-weight: ${({theme}) => theme.typography.variants.body.weight};
  letter-spacing: ${({theme}) => theme.typography.variants.body.letterSpacing};
  line-height: ${({theme}) =>
    pxToRem(
      theme.typography.variants.body.lineHeight,
      theme.typography.htmlFontSize
    )};
`

const smallMixin = <P extends Record<any, any>>(): FlattenInterpolation<
  ThemedStyledProps<P, DefaultTheme>
> => css<StyledProps<P>>`
  font-size: ${({theme}) =>
    pxToRem(
      theme.typography.variants.caption.size,
      theme.typography.htmlFontSize
    )};
  font-weight: ${({theme}) => theme.typography.variants.caption.weight};
  letter-spacing: ${({theme}) =>
    theme.typography.variants.caption.letterSpacing};
  line-height: ${({theme}) =>
    pxToRem(
      theme.typography.variants.caption.lineHeight,
      theme.typography.htmlFontSize
    )};

  b,
  strong {
    font-weight: ${({theme}) => theme.typography.fontWeights.boldest};
  }
`
export {
  h1Mixin,
  h2Mixin,
  h3Mixin,
  h4Mixin,
  h5Mixin,
  h6Mixin,
  bodyMixin,
  smallMixin
}
