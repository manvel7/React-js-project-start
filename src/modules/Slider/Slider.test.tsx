import React from "react"
import {render} from "@testing-library/react"
import {UiProvider} from "../../styles"
import Slider from "./Slider"

describe("Slider", () => {
  describe("Default variant", () => {
    it("should match snapshot", () => {
      const result = render(
        <UiProvider>
          <Slider
            slides={[
              <div key="1">first slide</div>,
              <div key="2">second slide</div>
            ]}
          />
        </UiProvider>
      )
      expect(result.container.firstChild).toMatchSnapshot()
    })
  })
})
