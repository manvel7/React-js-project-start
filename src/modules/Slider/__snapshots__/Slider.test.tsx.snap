// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider Default variant should match snapshot 1`] = `
.c6 {
  border: 0.2rem solid transparent;
  cursor: pointer;
  border-radius: 0.4rem;
  padding: 0.6rem 1.6rem;
  font-size: 1.6rem;
  font-weight: 600;
  -webkit-letter-spacing: 0em;
  -moz-letter-spacing: 0em;
  -ms-letter-spacing: 0em;
  letter-spacing: 0em;
  line-height: 1.6rem;
  background-color: transparent;
  color: #001254;
}

.c6 b,
.c6 strong {
  font-weight: 800;
}

.c7 {
  background-color: #F1F2F5;
  border: none;
  color: #001254;
  cursor: pointer;
  margin: 0 0.4rem;
  min-height: 3.2rem;
  min-width: 3.2rem;
  padding: 0.4rem 0.8rem;
  font-size: 1.8rem;
  font-weight: 600;
  -webkit-letter-spacing: 0.005em;
  -moz-letter-spacing: 0.005em;
  -ms-letter-spacing: 0.005em;
  letter-spacing: 0.005em;
  line-height: 2.4rem;
  border-radius: 10rem;
  background-color: #001254;
  color: #FEFEFE;
}

.c7[disabled] {
  cursor: initial;
}

.c7:first-child {
  margin-left: 0;
}

.c7:last-child {
  margin-right: 0;
}

.c7 b,
.c7 strong {
  font-weight: 800;
}

.c8 {
  background-color: #F1F2F5;
  border: none;
  color: #001254;
  cursor: pointer;
  margin: 0 0.4rem;
  min-height: 3.2rem;
  min-width: 3.2rem;
  padding: 0.4rem 0.8rem;
  font-size: 1.8rem;
  font-weight: 600;
  -webkit-letter-spacing: 0.005em;
  -moz-letter-spacing: 0.005em;
  -ms-letter-spacing: 0.005em;
  letter-spacing: 0.005em;
  line-height: 2.4rem;
  border-radius: 10rem;
}

.c8[disabled] {
  cursor: initial;
}

.c8:first-child {
  margin-left: 0;
}

.c8:last-child {
  margin-right: 0;
}

.c8 b,
.c8 strong {
  font-weight: 800;
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c3 {
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow: hidden;
  visibility: hidden;
}

.c3 > * {
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c1 {
  left: 50%;
  opacity: 0;
  position: absolute;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  -webkit-transition: opacity 0.2s ease-in-out;
  transition: opacity 0.2s ease-in-out;
  z-index: -1;
  opacity: 1;
  z-index: 1;
}

.c2 {
  left: 50%;
  opacity: 0;
  position: absolute;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  -webkit-transition: opacity 0.2s ease-in-out;
  transition: opacity 0.2s ease-in-out;
  z-index: -1;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-bottom: 1rem;
}

.c0 {
  padding: 3rem 0;
  position: relative;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div>
        first slide
      </div>
    </div>
    <div
      class="c2"
    >
      <div>
        second slide
      </div>
    </div>
    <div
      class="c3"
    >
      <div>
        first slide
      </div>
      <div>
        second slide
      </div>
    </div>
  </div>
  <div
    class="c4"
  >
    <div
      class="c5"
    >
      <button
        class="c6 hidden"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-angle-left fa-w-8 "
          data-icon="angle-left"
          data-prefix="fas"
          focusable="false"
          role="img"
          style="margin-right: 0.8rem;"
          viewBox="0 0 256 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
            fill="currentColor"
          />
        </svg>
        Previous
      </button>
      <button
        class="c7"
        disabled=""
      />
      <button
        class="c8"
      />
      <button
        class="c6"
      >
        Next
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-angle-right fa-w-8 "
          data-icon="angle-right"
          data-prefix="fas"
          focusable="false"
          role="img"
          style="margin-left: 0.8rem;"
          viewBox="0 0 256 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
